apps/dashboard/app/javascript/packs/files/datatable.js

```diff
@@ -1,112 +1,273 @@
-function update_datatables_status(api){
-  // from "function info ( api )" of https://cdn.datatables.net/select/1.3.1/js/dataTables.select.js
-  let rows    = api.rows( { selected: true } ).flatten().length,
-      page_info = api.page.info(),
-      msg = page_info.recordsTotal == page_info.recordsDisplay ? `Showing ${page_info.recordsDisplay} rows` : `Showing ${page_info.recordsDisplay} of ${page_info.recordsTotal} rows`;
+import 'datatables.net';
+import 'datatables.net-bs4/js/dataTables.bootstrap4';
+import 'datatables.net-select';
+import 'datatables.net-select-bs4';
+import Handlebars from 'handlebars';
+import { Swal } from './sweet_alert.js';
+import {} from './fileops.js';
+import {} from './uppy.js';
+import {} from './clipboard.js';
 
-  $('.datatables-status').html(`${msg} - ${rows} rows selected`);
-}
+let table = null;
+let onpopstate = null;
+let actionsBtnTemplate = null;
+let reportTransferTemplate = null;
 
-var table = $('#directory-contents').on('xhr.dt', function ( e, settings, json, xhr ) {
-  // new ajax request for new data so update date/time
-  if(json && json.time){
-    history.replaceState(_.merge({}, history.state, {currentDirectoryUpdatedAt: json.time}), null);
-  }
-}).DataTable({
-  autoWidth: false,
-  language: {
-    search: 'Filter:',
-  },
-  order: [[1, "asc"], [2, "asc"]],
-  rowId: 'id',
-  paging:false,
-  scrollCollapse: true,
-  select: {
-    style: 'os',
-    className: 'selected',
-    toggleable: true,
-    // don't trigger select checkbox column as select
-    // if you need to omit more columns, use a "selectable" class on the columns you want to support selection
-    selector: 'td:not(:first-child)'
-  },
-  // https://datatables.net/reference/option/dom
-  // dom: '', dataTables_info nowrap
-  //
-  // put breadcrmbs below filter!!!
-  dom: "<'row'<'col-sm-12'f>>" + // normally <'row'<'col-sm-6'l><'col-sm-6'f>> but we disabled pagination so l is not needed (dropdown for selecting # rows)
-       "<'row'<'col-sm-12'<'dt-status-bar'<'datatables-status float-right'><'transfers-status'>>>>"+
-       "<'row'<'col-sm-12'tr>>", // normally this is <'row'<'col-sm-5'i><'col-sm-7'p>> but we disabled pagination so have info take whole row
-  columns: [
-    {
-      data: null,
-      orderable: false,
-      defaultContent: '<input type="checkbox">',
-      render: function(data, type, row, meta) {
-        var api = new $.fn.dataTable.Api( meta.settings );
-        let selected = api.rows(meta.row, { selected: true }).count() > 0;
-        return `<input type="checkbox" ${selected ? 'checked' : ''}> ${selected ? 'checked' : ''}`;
-      }
+export { 
+  actionsBtnTemplate, reportTransferTemplate, table, dataFromJsonResponse, getEmptyDirs, getFilesAndDirectoriesFromDirectory, 

```

Comment:
I'd double check your exports. A quick spotcheck on the first one here `actionsBtnTemplate` doesn't seem to be used by any other file.